### ISSUE TITLE
Full network parameters not being updated in `run_bayesian_training()`

### ISSUE THREAD

**Issue by conorheins** [dimarkov/bllarse#29]
It appears in the following line 
https://github.com/dimarkov/bllarse/blob/c667b9eb18f2c179930c93503b86d26b49bb09d6/src/bllarse/utils.py#L419

we should be using `updated_nnet_params` as defined a few lines earlier
https://github.com/dimarkov/bllarse/blob/c667b9eb18f2c179930c93503b86d26b49bb09d6/src/bllarse/utils.py#L416

 not just the un-trained `nnet_params`. This matters specifically in the `full_network_finetuning.py` case where we are passing gradients back to the pretrained backbone network.

Am I misunderstanding something or is this a real bug?
